notary-api/src/jobs/paymentsHelpers.js

```diff
@@ -0,0 +1,23 @@
+import R from 'ramda';
+
+const reformatNotarizations = (result) => {
+  const reformat = R.pipe(
+    R.groupBy(R.prop('id')), // Object<BatPayId, Array<Pair<BatPayId,Address>>>
+    R.map(R.pluck('address')), // Object<BatPayId, Array<Address>>
+    R.mapObjIndexed(addresses => ({ [result]: addresses })),
+    // Object<BatPayId, Object<result,Array<Address>>>
+  );
+
+  return sellers => (sellers ? reformat(sellers) : {});
+};
+
+const mergeByBatPayId = (id, left, right) => ({ ...left, ...right });
+
+const reformatCompleted = reformatNotarizations('completed');
+const reformatRejected = reformatNotarizations('rejected');
+
+export const getResultsByBatPayId = (completedSellers, rejectedSellers) => {
+  const completed = reformatCompleted(completedSellers);
+  const rejected = reformatRejected(rejectedSellers);
+  return R.mergeWithKey(mergeByBatPayId, completed, rejected);
```

Comment:
At the end this line will be:
```suggestion
  return R.mergeWithKey(
    (id, completed, rejected) => ({ completed, rejected }),
    reformat(completedSellers),
    reformat(rejectedSellers));
```